Assignment4/MyLibrary/MyLibrary/SetWriter.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+
+namespace MyLibrary.Writer
+{
+    public sealed class SetWriter : IDisposable
+    {
+        private bool _DisposedValue;
+
+        private string? _filePath;
+
+        public string? FilePath
+        {
+            get => _filePath;
+            set => _filePath = value ??
+                throw new ArgumentNullException(nameof(value));
+        }
+
+        private StreamWriter? fileWriter;
+
+        public StreamWriter? FileWriter
+        {
+            get => fileWriter;
+            set => fileWriter = value ?? 
+                throw new ArgumentNullException(nameof(value));
+        }
```

Comment:
```suggestion
        private readonly StreamWriter _FileWriter;
```
You only need a readonly private variable here. It doesn't have to be marked as nullable because you should always be creating this in the constructor.